process/api.py

```diff
@@ -87,6 +99,161 @@ class CollectionViewSet(viewsets.ViewSetMixin, ListAPIView):
         "completed_at",
     ]
 
+    @csrf_exempt
+    @action(methods=["post"], detail=False)
+    def new(self, request):
+        input_message = request.data
+        if "source_id" not in input_message or "data_version" not in input_message:
+            return Response(
+                'Unable to parse input. Please provide {"source_id": "<source_id>", "data_version": "<data_version>"}',
+                status=status.HTTP_400_BAD_REQUEST,
+            )
```

Comment:
Use a serializer to handle this logic. See `CreateDatasetSerializer` in Pelican frontend. This would become:

```python
        serializer = CreateCollectionSerializer(data=request.data)
        serializer.is_valid(raise_exception=True)
```

Reply:
:+1 I wasn't sure how much of the original code I should change as part of this

Reply:
We can make these changes in this PR – or we can make this PR only about the new endpoint. But I think it's better to make a full transition to DjangoRESTFramework rather than going halfway. Up to you whether you want to copy-paste the changes related to old endpoints into a separate branch and focus on just the new endpoint here.